static/app/components/forms/teamSelector.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+import {addTeamToProject} from 'app/actionCreators/projects';
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import SelectControl, {ControlProps} from 'app/components/forms/selectControl';
+import {IconAdd, IconUser} from 'app/icons';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {Organization, Project, Team} from 'app/types';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+import withTeams from 'app/utils/withTeams';
+
+import IdBadge from '../idBadge';
+import Tooltip from '../tooltip';
+
+const UnassignedWrapper = styled('div')`
+  display: flex;
+  align-items: center;
+`;
+
+const StyledIconUser = styled(IconUser)`
+  margin-left: ${space(0.25)};
+  margin-right: ${space(1)};
+  color: ${p => p.theme.gray400};
+`;
+
+// An option to be unassigned on the team dropdown
+const unassignedOption = {
+  value: null,
+  label: (
+    <UnassignedWrapper>
+      <StyledIconUser size="20px" />
+      {t('Unassigned')}
+    </UnassignedWrapper>
+  ),
+  searchKey: 'unassigned',
+  actor: null,
+  disabled: false,
+};
+
+// Ensures that the svg icon is white when selected
+const unassignedSelectStyles = {
+  option: (provided, state: any) => ({
```

Comment:
Is there a better type than `any` we can use here? I think react select does have state types

Reply:
hmmm can't seem to find any :/ seems like the `styleFn` definition in @types/react-select is just 
`export type styleFn = (base: CSSProperties, state: any) => CSSProperties;`

might be looking in the wrong place tho

Reply:
👍 nvm you good then